File: Source/Back-end/src/main/java/org/elis/manutenzione/dto/response/AccettaManutenzioneResponse.java
package org.elis.manutenzione.dto.response;

import lombok.Data;
import java.time.LocalDate;

/**
 * DTO che rappresenta la risposta alla richiesta di accettazione di una manutenzione.
 */
@Data
public class AccettaManutenzioneResponse {

    /**
     * L'ID della manutenzione accettata.
     */
    private long id;

    /**
     * La descrizione della manutenzione.
     */
    private String descrizione;

    /**
     * Il nome del luogo in cui si effettua la manutenzione.
     */
    private String luogo;

    /**
     * Il nucleo a cui appartiene il luogo.
     */
    private String nucleo;

    /**
     * La priorità della manutenzione.
     */
    private int priorita;

    /**
     * La data prevista per l'esecuzione della manutenzione.
     */
    private LocalDate dataPrevista;

    /**
     * Il nome della manutenzione.
     */
    private String nome;
}

File: Source/Front-end/front-end/src/app/model/response/accetta-manutenzione-response.ts
export class AccettaManutenzioneResponse {
  id: number;
  descrizione: string;
  luogo: string;
  nucleo: string;
  priorita: number;
  dataPrevista: Date;
  nome: string;

  constructor(id: number, descrizione: string, luogo: string, nucleo: string, priorita: number, dataPrevista: Date, nome: string) {
    this.id = id;
    this.descrizione = descrizione;
    this.luogo = luogo;
    this.nucleo = nucleo;
    this.priorita = priorita;
    this.dataPrevista = dataPrevista;
    this.nome = nome;
  }
}
